src/lib/utils/track/index.tsx

```diff
@@ -90,7 +88,7 @@ export interface Track<P = any, S = null, T extends Schema.Global = Schema.Entit
 export const track: Track = (trackingInfo, options) => {
   return _track(trackingInfo, {
     ...options,
-    dispatch: (data) => postEvent(data),
+    dispatch: (data) => postEventToProviders(data),
```

Comment:
here's the main change. we don't just post to native anymore. we post to providers. on ios we add the native analytics as a provider, on android we add segment-rn. we could make the logging into a provider as well, etc.

Reply:
nonblocking: I feel like the word 'provider' here is a bit overloaded with react providers. A more standard pattern for this kind of thing might be to have a single EventEmitter for analytics events, and then to add eventListeners for each thing you  currently call a 'provider'.

```ts
const analyticsEvents = new EventEmitter<TypesLol>()
if (Platform.OS === 'ios') {
  analyticsEvents.addEventListener('event', (data) => NativeModules.Blah.postEvent(data))
} else {
  analyticsEvents.addEventListener('event', (data) => trackSegmentEvent(data))
  analyticsEvents.addEventListener('identify', (data) => trackSegmentIdentify(data))
}
if (__DEV__) {
  analyticsEvents.addEventListener('event', (data) => console.log('[Event tracked]', data))
}
// .. etc
```

and then this line would be 
```suggestion
    dispatch: (data) => analyticsEvents.emit('event', data),
```

It's OK how it is though

Reply:
huh. interesting. the reverse way. emit events, and whoever cares can listen.